code/datums/station_traits/job_traits.dm

```diff
@@ -93,3 +110,58 @@
 	// monkey carries the crates, the age of robot is over
 	if(GLOB.cargo_ripley)
 		qdel(GLOB.cargo_ripley)
+
+/datum/station_trait/job/bridge_assistant
+	name = "Bridge Assistant"
+	button_desc = "Sign up to become the Bridge Assistant and watch over the Bridge."
+	weight = 2
+	report_message = "We have installed a Bridge Assistant on your station."
+	show_in_report = TRUE
+	can_roll_antag = CAN_ROLL_PROTECTED
+	job_to_add = /datum/job/bridge_assistant
+
+/datum/station_trait/job/bridge_assistant/New()
+	. = ..()
+	RegisterSignal(SSatoms, COMSIG_SUBSYSTEM_POST_INITIALIZE, PROC_REF(add_coffeemaker))
+
+/datum/station_trait/job/bridge_assistant/on_lobby_button_update_overlays(atom/movable/screen/lobby/button/sign_up/lobby_button, list/overlays)
+	. = ..()
+	overlays += "bridge_assistant"
+
+/// Creates a coffeemaker in the bridge, if we don't have one yet.
+/datum/station_trait/job/bridge_assistant/proc/add_coffeemaker(datum/source)
+	SIGNAL_HANDLER
+	var/area/bridge = GLOB.areas_by_type[/area/station/command/bridge]
+	if(isnull(bridge)) //no bridge, what will he assist?
+		return
+	var/list/possible_coffeemaker_positions = list(/area/station/command/bridge, /area/station/command/meeting_room)
+	for(var/possible_position in possible_coffeemaker_positions) //don't spawn a coffeemaker if there is already one on the bridge
+		var/area/possible_area = GLOB.areas_by_type[possible_position]
+		if(possible_area && (locate(/obj/machinery/coffeemaker) in possible_area))
+			return
```

Comment:
Perhaps you should use `SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/coffeemaker)` and checking the area of those rather than using `locate() in area`, since it likely is as bad as iterating over area contents

![image](https://github.com/tgstation/tgstation/assets/51863163/7aa9cba2-1ca1-424a-b039-98db3a272031)